### DISCUSSION TITLE
How to get "temperature", "time" and "time step size"

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#17528] [Q&A General] [answered]
All,

I am solving a diffusion equation with energy equation (temperature).
I am writing a C++ class that inherits from Material. My program looks like the one below,
where "time" is the computation time, "del_time" is the time step size.
Please let me know how to get "temperature," "time" and "time step size."

float base, temp, flux, del_time, temperature;
float constant_A = 10, constant_B = 100, constant_C = 1000;

If (time <= 0)
base = 0;
if (temperature <= 100)
temp = constant_A * temperature;
else
temp = constant_B * temperature;
    
base = base + (temp * del_time);
flux = constant_C / base;

Sincerely,
Thomas

**Comment by aeslaughter** [comment, marked as answer]
You would get temperature using coupling as in: https://mooseframework.inl.gov/getting_started/examples_and_tutorials/examples/ex03_coupling.html.

The time and timestep are available as _t and _dt, see https://mooseframework.inl.gov/docs/doxygen/moose/classMaterial.html
